crates/cast/src/starknet_commands/show_config.rs

```diff
@@ -0,0 +1,26 @@
+use anyhow::Result;
+use camino::Utf8PathBuf;
+use cast::helpers::response_structs::ShowConfigResponse;
+use cast::helpers::scarb_utils::CastConfig;
+use clap::Args;
+
+#[derive(Args)]
+#[command(about = "Show current configuration being used", long_about = None)]
+pub struct ShowConfig {}
+
+#[allow(clippy::ptr_arg)]
+pub async fn show_config(
+    cast_config: CastConfig,
+    profile: String,
+    scarb_path: Utf8PathBuf,
+    network: String,
+) -> Result<ShowConfigResponse> {
+    Ok(ShowConfigResponse {
+        profile,
+        scarb_path,
+        network,
+        rpc_url: cast_config.rpc_url,
+        account: cast_config.account,
+        account_file_path: cast_config.accounts_file,
+    })
+}
```

Comment:
Let's also add some tests :) After they're added I will access the pipeline to go through

Reply:
before adding test, i wanted to know your thoughts on:
> current its printing fields that are not present (empty strings) as 0x0 due to the way print_command_result works, should i create a new duplicate function to not print this if they are not set?

Reply:
Ideally, the current function should be modified to not print these fields. How much work do you think this would be?

Reply:
that function calls another function (`print_formatter`) that does this conversion, we can pass in a boolean flag depending on which it can choose to convert values to numeric. does that sound good?

https://github.com/foundry-rs/starknet-foundry/blob/302627ec8ef83bd8d80626d143a34f794d3a95dc/crates/cast/src/lib.rs#L308-L325

Reply:
How about passing `Options` as `ShowConfigResponse` fields? We could modify the print_formatted function to not print `None`'s. wdyt?

Reply:
thats what i tried initially, but `print_command_result` was panicking somwhere here: https://github.com/foundry-rs/starknet-foundry/blob/302627ec8ef83bd8d80626d143a34f794d3a95dc/crates/cast/src/lib.rs#L361

when value is `None` i think `as_str` returns `None` https://docs.rs/serde_json/1.0.107/serde_json/value/enum.Value.html#method.as_str

Reply:
Any updates on `print_command_result` are welcome, so feel free to do it. 

> Is it a good idea to ignore value's that are not convertable to string?

Are there any?

Reply:
> Are there any?

Not atm but in future we might want to print more complicated nested values.

considering that i think for now adding a boolean to not convert values to numeric make more sense.

Reply:
> How about passing Options as ShowConfigResponse fields? We could modify the print_formatted function to not print None's. wdyt?

that would be a way to go imo

> actually then we would print empty values for those, so would need to remove it somehow so would have to make it optional and ignore None values

I think we should just not print fields that are None. Not sure their omission would make debugging harder - in contrary, you know they were just empty (not set at all) so it is a hint that maybe something went not as you expected it to. what do you think?

Reply:
yeah i think making it optional and removing None make sense, will implement that.

Reply:
fixed this, moving conversation to new thread